scanpy/_metadata.py

```diff
@@ -1,14 +1,33 @@
+import traceback
 from pathlib import Path
 
 here = Path(__file__).parent
 
+
+def refresh_entry_points():
+    """\
+    Under some circumstances, (e.g. when installing a PEP 517 package via pip),
+    pkg_resources.working_set.entries is stale. This tries to fix that.
+    """
+    try:
+        import sys
+        import pkg_resources
+
+        ws: pkg_resources.WorkingSet = pkg_resources.working_set
+        for entry in sys.path:
+            ws.add_entry(entry)
+    except Exception:
+        pass
```

Comment:
Can you link to an upstream issue or documentation here?

Reply:
Will do. In short: `setuptools_scm` shouldn’t use `pkg_resources` anymore: pypa/setuptools_scm#513